### PR TITLE
Configurable snapping

### PR DESCRIPTION
**Pull request by TheMarex**
This makes it configurable from profiles whether to not snap to road or not. This is useful for ferries and the like where snapping makes no sense.

- [x] Rebase on develop
- [x] Write test to verify the behaviour
- [x] Fix routability cucumber tests that rely on snapping to ferry roads _doesn't seem to be an issue for the car profile_

### CODE REVIEW

File: data_structures/node_based_graph.hpp [hunk 1]
```diff
@@ -60,6 +60,7 @@ struct NodeBasedEdgeData
     bool access_restricted : 1;
     bool reversed : 1;
     bool roundabout : 1;
+    bool startpoint : 1;
     TravelMode travel_mode : 4;
```
Comment (MoKob) [inline data_structures/node_based_graph.hpp]: Is this struct defined at multiple locations? Would prefer to move it to an include and only define it once.

File: data_structures/import_edge.hpp [hunk 2]
```diff
@@ -55,6 +56,7 @@ struct NodeBasedEdge
     bool backward : 1;
     bool roundabout : 1;
     bool access_restricted : 1;
+    bool startpoint : 1;
     bool is_split : 1;
     TravelMode travel_mode : 4;
 };
```
Comment (MoKob) [inline data_structures/import_edge.hpp]: multiply defined? Just missing is_split here?

File: extractor/edge_based_graph_factory.cpp [hunk 3]
```diff
@@ -78,6 +78,11 @@ void EdgeBasedGraphFactory::GetEdgeBasedNodes(std::vector<EdgeBasedNode> &nodes)
     nodes.swap(m_edge_based_node_list);
 }
 
+void EdgeBasedGraphFactory::GetStartPointMarkers(std::vector<bool> &node_is_startpoint)
+{
+    m_edge_based_node_is_startpoint.swap(node_is_startpoint);
+}
```
Comment (MoKob) [inline extractor/edge_based_graph_factory.cpp]: Effective C++, Item 25 (https://books.google.de/books?id=U7lTySXdFk0C&pg=PT139&lpg=PT139&dq=prefer+std::swap+over+member&source=bl&ots=nO6Jrdv9kb&sig=nX3EPlZUTHahOvHTCCSoSba8RmM&hl=en&sa=X&ved=0ahUKEwjCzfvp_trJAhVE_Q4KHRZpA4YQ6AEIOjAF#v=onepage&q=prefer%20std%3A%3Aswap%20over%20member&f=false).

You should consider adding `using std::swap` and call `swap(node_is_starpoint, m_edge_base_node_is_startpoint)` as a general way to invoke swap.

File: extractor/extractor.cpp [hunk 4]
```diff
@@ -529,32 +531,31 @@ extractor::BuildEdgeExpandedGraph(std::vector<QueryNode> &internal_to_external_n
...
-            config.edge_segment_lookup_path,
-            config.edge_penalty_path,
-            config.generate_edge_lookup
+                                 config.edge_segment_lookup_path, config.edge_penalty_path,
+                                 config.generate_edge_lookup
 #ifdef DEBUG_GEOMETRY
-            , config.debug_turns_path
+                                 ,
+                                 config.debug_turns_path
 #endif
```
Comment (MoKob) [inline extractor/extractor.cpp]: why in two lines? Would prefer consistent style with ',' at begin of the line

File: extractor/extractor.cpp [hunk 5]
```diff
@@ -572,26 +573,44 @@ void extractor::WriteNodeMapping(const std::vector<QueryNode> & internal_to_exte
 
     Saves tree into '.ramIndex' and leaves into '.fileIndex'.
  */
-void extractor::BuildRTree(const std::vector<EdgeBasedNode> &node_based_edge_list,
+void extractor::BuildRTree(std::vector<EdgeBasedNode> node_based_edge_list,
+                           std::vector<bool> node_is_startpoint,
                            const std::vector<QueryNode> &internal_to_external_node_map)
```
Comment (MoKob) [inline extractor/extractor.cpp]: We are not taking ownership of these vectors. So sink parameter passing does not apply. Shouldn't these be passed by const reference?

File: data_structures/node_based_graph.hpp (hunk 1 above)
Comment (TheMarex) [inline data_structures/node_based_graph.hpp]: It's not the struct, just shares the same member names for consistency.

File: extractor/extractor.cpp (hunk 5 above)
Comment (TheMarex) [inline extractor/extractor.cpp]: Needs to be sink because we filter it, look in the diff a few lines down.

**Pull request merged**